ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -143,13 +143,23 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
         }
     }
 
-    public async showNodePicker(expectedContextValue: string): Promise<IAzureNode> {
-        const picks: PickWithData<SubscriptionNode>[] = this._subscriptionNodes.map((n: SubscriptionNode) => new PickWithData<SubscriptionNode>(n, n.treeItem.label, n.subscription.subscriptionId));
-        let node: AzureNode = (await this._ui.showQuickPick<SubscriptionNode>(picks, localize('selectSubscription', 'Select a Subscription'))).data;
+    public async showNodePicker(expectedContextValues: string | string[], startingNode?: IAzureNode): Promise<IAzureNode> {
+        if (!Array.isArray(expectedContextValues)) {
+            expectedContextValues = [expectedContextValues];
+        }
+
+        let node: IAzureNode;
+        if (startingNode) {
+            node = startingNode;
+        } else {
+            let picks: PickWithData<IAzureNode>[] = this._subscriptionNodes.map((n: SubscriptionNode) => new PickWithData(n, n.treeItem.label, n.subscription.subscriptionId));
+            picks = picks.concat(this._rootNodes.filter((n: AzureNode) => n.includeInNodePicker(<string[]>expectedContextValues)).map((n: AzureNode) => new PickWithData(n, n.treeItem.label)));
+            node = (await this._ui.showQuickPick<IAzureNode>(picks, localize('selectSubscription', 'Select a Subscription'))).data;
+        }
 
-        while (node.treeItem.contextValue !== expectedContextValue) {
+        while (!expectedContextValues.some((val: string) => node.treeItem.contextValue === val)) {
```

Comment:
Is this different from a `while(expectedContextValues.indexOf(node.treeItem.contextValue) === -1)` ?

Reply:
Those look pretty much the same to me